src/DrDotnet/Services/ProcessDiscovery.cs

```diff
@@ -44,43 +42,29 @@ private bool TryGetManagedAssemblyNameFromPid(int pid, out string assemblyName,
         }
     }
 
-    public async ValueTask<List<ProcessInfo>> GetDotnetProcessesAsync(Action<float> progressCallback)
+    public List<ProcessInfo> GetDotnetProcesses(Action<float> progressCallback)
     {
         _logger.LogInformation("Listing dotnet processes...");
 
         var dotnetProcesses = new List<ProcessInfo>();
 
-        // Todo: Use IAsyncEnumerable
-        
         try
         {
-            var currentProcess = Process.GetCurrentProcess();
             var processes = DiagnosticsClient.GetPublishedProcesses().ToArray();
             
             for (int i = 0; i < processes.Length; i++)
             {
                 progressCallback(1f * i / processes.Length);
             
-                try
-                {
-                    Process process = Process.GetProcessById(processes[i]);
-
-                    if (processes[i] == currentProcess.Id)
-                        continue;
-
-                    _logger.LogInformation($"- [Process] Id: {processes[i]}, Name: {process.ProcessName}");
-
-                    if (!TryGetManagedAssemblyNameFromPid(process.Id, out string assemblyName, out string version))
-                    {
-                        continue;
-                    }
-
-                    dotnetProcesses.Add(new ProcessInfo { Pid = process.Id, Name = process.ProcessName, StartTime = process.StartTime, ManagedAssemblyName = assemblyName, Version = version });
-                }
-                catch (Exception e)
+                _logger.LogInformation($"- [Process] Id: {processes[i]}");
+                
+                if (!TryGetManagedAssemblyNameFromPid(processes[i], out string assemblyName, out string version))
                 {
-                    _logger.LogError(e, "Can't read process {ProcessId} information.", processes[i]);
+                    dotnetProcesses.Add(new ProcessInfo { Pid = processes[i], ManagedAssemblyName = "unknown", Version = "unknown" });
```

Comment:
I think we should skip processes for which we failed to get the ManagedAssemblyName... if I'm not mistaken we always fail to attach to those processes

Reply:
Indeed ! I'll keep the log however, it can be useful in case you don't see your process listed at least know if the socket is visible